src/base/bittorrent/session.cpp

```diff
@@ -1353,6 +1354,8 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
         settingsPack.set_int(libt::settings_pack::mixed_mode_algorithm, libt::settings_pack::peer_proportional);
         break;
     }
+    settingsPack.set_bool(libt::settings_pack::enable_incoming_tcp, !(isUTPEnabled() && isUtpOnlyModeEnabled()));
```

Comment:
I can't find code that enables/disables cbUtpOnly in the settings dialog when isUTPEnabled() changes.

Reply:
since uTPOnlyMode is in advanced settings, users should know what they are messing with, I don't think additional mechanism is necessary.